### ISSUE TITLE
false positive with `link` no-deprecated

### ISSUE THREAD

**Issue by Mister-Hope** [oxc-project/tsgolint#1046]
<img width="906" height="1225" alt="Image" src="https://github.com/user-attachments/assets/eea8ee9d-6352-4aed-8544-0a40018ce205" />

Any `link` property in plain object which passes to another function is now incorrectly reported with `typescript(no-deprecated): `link` is deprecated. A legacy feature for browser compatibility`.

A sample code below:

```ts
import { sidebar } from "vuepress-theme-hope";

export const enSidebar = sidebar({
  "/": [
    "",
    {
      text: "Demo",
      icon: "laptop-code",
      prefix: "demo/",
      link: "demo/",
      children: "structure",
    },
    {
      text: "Articles",
      icon: "book",
      prefix: "posts/",
      children: "structure",
    },
    "intro",
    "slide",
  ],
});
```
